src/sentry/seer/similarity/utils.py

```diff
@@ -266,31 +230,52 @@ def _process_frames(frames: list[dict[str, Any]]) -> list[str]:
             continue
 
         # For each exception, extract its type, value, and up to limit number of stacktrace frames
-        exc_type, exc_value, frame_strings = "", "", []
+        exc_type, exc_value = "", ""
+        frame_strings: list[str] = []
         if exception_type == "stacktrace":
-            frame_strings = _process_frames(exception.get("values", []))
+            frame_strings, processing_metrics = _process_frames(
+                exception.get("values", []), processing_metrics
+            )
         else:
             for exception_value in exception.get("values", []):
                 if exception_value.get("id") == "type":
                     exc_type = _get_value_if_exists(exception_value)
                 elif exception_value.get("id") == "value":
                     exc_value = _get_value_if_exists(exception_value)
-                elif exception_value.get("id") == "stacktrace" and frame_count < MAX_FRAME_COUNT:
-                    frame_strings = _process_frames(exception_value["values"])
+                elif (
+                    exception_value.get("id") == "stacktrace"
+                    and processing_metrics["frame_count"] < MAX_FRAME_COUNT
+                ):
+                    frame_strings, processing_metrics = _process_frames(
+                        exception_value["values"], processing_metrics
+                    )
         if (
             platform not in SYSTEM_FRAME_CHECK_BLACKLIST_PLATFORMS
-            and is_frames_truncated
+            and processing_metrics["is_frames_truncated"]
             and not app_hash
         ):
             raise TooManyOnlySystemFramesException
-        if has_no_filename_or_module:
+        if processing_metrics["has_no_filename_or_module"]:
             raise NoFilenameOrModuleException
         # Only exceptions have the type and value properties, so we don't need to handle the threads
         # case here
         header = f"{exc_type}: {exc_value}\n" if exception["id"] == "exception" else ""
 
         result_parts.append((header, frame_strings))
 
+    return generate_stacktrace_string(
```

Comment:
All of this code has been moved within the function:
<img width="666" alt="image" src="https://github.com/user-attachments/assets/061d868f-d891-42ed-9df1-cf85fd3abbfe">